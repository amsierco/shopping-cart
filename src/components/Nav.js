import { Link } from "react-router-dom";
import React from "react";

const Nav = () => {
    return (
        <div className="navbar">
            <div className="logo">
                Fancy Logo
            </div>
            <nav>
                <Link to="/">Home</Link>
                <Link to="/products">Products</Link>
                <Link to="/">Cart</Link>
            </nav>
        </div>
    );
}

export default Nav;