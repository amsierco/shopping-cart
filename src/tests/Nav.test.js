import { render, screen } from '@testing-library/react';
import userEvent from "@testing-library/user-event";
import { BrowserRouter, Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import Nav from '../components/Nav';

describe('Nav component', () => {
    it("renders home nav link", () => {
        const { container } = render(<Nav />, { wrapper: BrowserRouter });
        expect(container).toMatchSnapshot();
    });

    it("renders home page after nav button click", () => {
        const history = createMemoryHistory({ initialEntries: ["/"] });

        render(
            <Router location={history.location} navigator={history}>
                <Nav />
            </Router>
        );

        const homeLink = screen.getByRole("link", { name: "Home" });
        userEvent.click(homeLink);

        expect(history.location.pathname).toBe('/');
    });

    it("renders products page after nav button click", () => {
        const history = createMemoryHistory({ initialEntries: ["/"] });

        render(
            <Router location={history.location} navigator={history}>
                <Nav />
            </Router>
        );

        const homeLink = screen.getByRole("link", { name: "Products" });
        userEvent.click(homeLink);

        expect(history.location.pathname).toBe('/products');
    });
});
