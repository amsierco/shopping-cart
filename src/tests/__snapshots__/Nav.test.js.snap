// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component renders home nav link 1`] = `
<div>
  <div
    class="navbar"
  >
    <div
      class="logo"
    >
      Fancy Logo
    </div>
    <nav>
      <a
        href="/"
      >
        Home
      </a>
      <a
        href="/products"
      >
        Products
      </a>
      <a
        href="/"
      >
        Cart
      </a>
    </nav>
  </div>
</div>
`;
